src/utils.rs

```diff
@@ -159,6 +159,27 @@ macro_rules! impl_enum_decodable {
     };
 }
 
+// Same as try!, but for Option
+#[macro_export]
+macro_rules! try_opt {
+    ($expr:expr) => (match $expr {
+        Some(val) => val,
+        None => { return None; }
+    })
+}
+
+// Unwraps an iterator of Options. If any element is None, return None
+#[macro_export]
+macro_rules! try_opt_collect {
+    ($expr:expr) => ({
+        let mut res = Vec::with_capacity($expr.size_hint().0);
```

Comment:
I didn't found this impl of IntoIterator for Option. That's exactly what I need, thanks for the pointer.